tests/test_intervention_outcome.py

```diff
@@ -0,0 +1,160 @@
+from conftest import *
+from datetime import datetime
+
+from models.appendix import InterventionReason
+from models.prescription import Intervention
+from security.role import Role
+
+import json
+from collections.abc import Mapping
+
+# import auxiliary dictionaries to this test
+from tests.aux_test_intervention import *
+
+
+def get_nested_value(data, path):
+    keys = path.split(".")
+    for key in keys:
+        if isinstance(data, Mapping) and key in data:
+            data = data[key]
+        elif isinstance(data, list) and key.isdigit():
+            data = data[int(key)]
+        else:
+            return None
+    return data
+
+
+def compare_specific_fields(data1, data2, fields_to_compare):
+    differences = {}
+
+    for field in fields_to_compare:
+        value1 = get_nested_value(data1, field)
+        value2 = get_nested_value(data2, field)
+
+        if value1 is None and value2 is None:
+            continue
+        elif value1 is None:
+            differences[field] = ("Field not present in first JSON", value2)
+        elif value2 is None:
+            differences[field] = (value1, "Field not present in second JSON")
+        elif value1 != value2:
+            differences[field] = (value1, value2)
+
+    return differences
+
+
+def compare_json_fields(json_str1, json_str2, fields_to_compare):

```

Comment:
Existem algumas libs que fazem essa comparação. Talvez seja mais simples.
Pode dar uma olhada nessas:
https://github.com/seperman/deepdiff
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertDictEqual

*sugestão. Acho que pode deixar o código mais limpo

Reply:
deepiff does have a parameter for defining only some fields for the operation , but for a deep nested json, it is more difficult to define
assertDictEqual does not have a parameter for defining only some fields for the operation

Conclusion: the current method is quite straightforward, have a good visibility for the nested json - and works !

Reply:
ok!
Para conhecimento, o deepdiff tem um método que retorna as diferenças entre os dicts. Você acabaria tendo que testar somente se os atributos diferentes são aceitáveis. Seria uma lista com os atributos de data e id, basicamente.